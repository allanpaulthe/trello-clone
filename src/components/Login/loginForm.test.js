import React from 'react';
import LoginForm from './loginForm';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('login', () => {
    it('renders correctly', () => {
        const output = shallow(
            <LoginForm.WrappedComponent />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('should handle wrong form submission', () => {
        const output = shallow(
            <LoginForm.WrappedComponent />
        );
        const event = { preventDefault: () => false };
        expect(output.state().wrong).toEqual(false)
        const link = output.find('input').get(2);
        link.props.onClick(event);
        expect(output.state().wrong).toEqual(true);
    });
    it('should handle correct form submission', () => {
        const output = shallow(
            <LoginForm.WrappedComponent />
        );
        const event = { preventDefault: () => false };
        output.setState({
            email:'alan@gmail.com',
            password:'alan'
        })
        const link = output.find('input').get(2);
        link.props.onClick(event);
        expect(output.state().wrong).toEqual(false);
    });

});