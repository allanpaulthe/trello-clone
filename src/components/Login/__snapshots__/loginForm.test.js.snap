// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders correctly 1`] = `
<form
  action=""
>
  <div
    className="loginForm"
  >
    <p>
      Email 
      <span>
        (or username)
      </span>
       
    </p>
    <input
      id="email"
      onChange={[Function]}
      placeholder="e.g., e.hughes@downton.org.uk"
      required={true}
      type="text"
      value=""
    />
    <p>
      Password
    </p>
    <input
      id="password"
      onChange={[Function]}
      placeholder="e.g., ••••••••••••"
      required={true}
      type="password"
    />
    <input
      className="button green-button"
      onClick={[Function]}
      type="submit"
      value="Log In"
    />
  </div>
</form>
`;
